Replace 'Maybe Bool'  with a dedicated sum-type.

KtorZ opened the pull request:
# Issue Number

<!-- Put here a reference to the issue this PR relates to and which requirements it tackles -->

#1701

# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- 9aaf4ad62c6b5982ef381073f84f2b76d5251f2f
  :round_pushpin: **log when no migration is needed**
  
- 7c3d459adfbd86353bd84b31f8d9220bd9c9532f
  :round_pushpin: **replace 'Maybe Bool' with a dedicated sum-type for more clarity**
  
# Comments

<!-- Additional comments or screenshots to attach if any -->

Caught me one time already... Won't caught anyone else. 

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Assign the PR to a corresponding milestone
 ✓ Acknowledge any changes required to the Wiki
-->

KtorZ committed:
log when no migration is needed

KtorZ committed:
replace 'Maybe Bool' with a dedicated sum-type for more clarity

jonathanknowles commented on lib/core/src/Cardano/Wallet/DB/Sqlite.hs:
```diff
@@ -306,6 +306,13 @@ findDatabases tr dir = do
+data SQLColumnStatus
+    = TableDoesntExist
+    | ColumnDoesntExist
+    | ColumnExists
```
Perhaps this wording would be a little more concise:
```suggestion
    = TableMissing
    | ColumnMissing
    | ColumnPresent
```

jonathanknowles commented on lib/core/src/Cardano/Wallet/DB/Sqlite.hs:
```diff
@@ -306,6 +306,13 @@ findDatabases tr dir = do
 
+-- | A data-type for capturing column status. Used to be represented as a
+-- 'Maybe Bool' which is somewhat confusing to interpret.
+data SQLColumnStatus
```
Suggestion: use PascalCase here, treating SQL as a word.

```suggestion
data SqlColumnStatus
```

Similar examples:
- [SqlBackend](https://hackage.haskell.org/package/persistent-2.10.5.1/docs/Database-Persist-Sql.html#t:SqlBackend)
- [InsertSqlResult](https://hackage.haskell.org/package/persistent-2.10.5.1/docs/Database-Persist-Sql.html#t:InsertSqlResult)

KtorZ replied:
I do actually prefer the golang guidelines on acronyms, but whatever ^.^

jonathanknowles commented on lib/core/src/Cardano/Wallet/DB/Sqlite.hs:
```diff
@@ -306,6 +306,13 @@ findDatabases tr dir = do
   where
     expectedPrefix = T.pack $ keyTypeDescriptor $ Proxy @k
 
+-- | A data-type for capturing column status. Used to be represented as a
+-- 'Maybe Bool' which is somewhat confusing to interpret.
```
Suggestion: remove the line starting with "Used to be represented...".

Justification: the new code is already very clear. :+1: 

(We don't need to remind the reader how confusing it used to be :smile_cat:)

```suggestion
-- | Indicates whether or not a column and its parent table are present
--   in the database.
```

KtorZ committed:
create a 'passphraseScheme' column when it's missing

Unfortunately, the manual migration are applied before the automated
migration. Yet, the passphraseScheme migration requires the column to
already exists in order to populate it. Therefore, we make sure to first
create it, and then, can safely assign it a value. If it already exists,
we simply assign a value.

piotr-iohk reviewed (approved):
works on my computer :tada:
